api_pagopa.yaml

```diff
@@ -0,0 +1,57 @@
+swagger: "2.0"
+info:
+  version: 1.0.0
+  title: Proxy API
+  description: Mobile and web proxy API gateway.
+host: localhost
+basePath: /api/v1
+schemes:
+  - https
+security:
+  - Token: []
+paths:
+  "/user":
```

Comment:
`user` looks a bit too generic here, this is not really a "backend" user, perhaps `pagopaUser` ? or `pagopa-user` ?

or perhaps we should mount the API on `/pagopa/api/v1` ?

Reply:
I've choose to mount the route on /pagopa/api/v1

Reply:
looking at this definition a second time, is the lookup of the used done through the auth token? I think conceptually the token used for lookup by pagopa is a sort of ID: pagopa is asking us "give me the user that correspond to this token?"

I think that the endpoint should be in the for `/user/:session_token`, and we should add a 404 to the responses (when token does not exist).

And we also should keep the auth token in the headers, that would be used by the pagopa backend to authenticate to our backend.

makes sense?

Reply:
yes this is a WIP, the final authentication is not in place yet.

As we said in the story [here](https://www.pivotaltracker.com/story/show/157719534), we should not use the same authentication token used by the APP (otherwise the PagoPa could operate as the APP itself). We need a way to restrict the operations a client can do when he uses the /pagopa/api/v1 endpoint, or we need to generate a secondary token only used by the PagoPa client.

Reply:
Yes indeed we need to give pagopa e different token than the one used by the app with the backend (sorry if it was confusing, by `session_token` I was meaning a token for the app/pagopa session). The safest path would be to generate a 2nd token each time we generate the app/backend session token and give this 2nd token to the app, together with the backend session token. Thus the app will have to keep both tokens. (cc @fpersico @fgiobergia ).
On the backend we need to keep a mapping from `pagopa_session_token` to `backend_session_token`.